test/envoye2e/env/setup.go

```diff
@@ -386,43 +412,52 @@ func (s *TestSetup) unmarshalStats(statsJSON string) map[string]int {
 	return statsMap
 }
 
-// VerifyStats verifies Envoy stats.
-func (s *TestSetup) VerifyStats(expectedStats map[string]int, port uint16) {
-	s.t.Helper()
-
-	check := func(actualStatsMap map[string]int) error {
-		for eStatsName, eStatsValue := range expectedStats {
-			aStatsValue, ok := actualStatsMap[eStatsName]
-			if !ok && eStatsValue != 0 {
-				return fmt.Errorf("failed to find expected stat %s", eStatsName)
-			}
-			if aStatsValue != eStatsValue {
-				return fmt.Errorf("stats %s does not match. expected vs actual: %d vs %d",
-					eStatsName, eStatsValue, aStatsValue)
-			}
+// unmarshalPrometheusStats Unmarshals Envoy Prometheus stats from file format into a map, where stats name is
+// key, and stats value is value. Currently, this is good for counters and not much for histograms.
+func (s *TestSetup) unmarshalPrometheusStats(statsStr string) map[string]int {
```

Comment:
Is there any prometheus library to do stats parsing? e.g. https://github.com/prometheus/common/blob/master/expfmt/text_parse.go

Reply:
Thanks for the pointer.. using this..